### PR TITLE
Add a timeout to the url requests

### PR DESCRIPTION
**Pull request by Jaimepas77**
Just added a timeout to the url request so when it's not answered it doesn't wait a long time before throwing an error.

### CODE REVIEW

File: tldr.py [hunk 1]
```diff
@@ -160,6 +160,7 @@ def get_page_for_platform(
         try:
             data = urlopen(
                 Request(page_url, headers=REQUEST_HEADERS),
+                timeout = 2,
```
Comment (MasterOdin) [inline tldr.py:163]:
```suggestion
                timeout=2,
```

File: tldr.py [hunk 2]
```diff
@@ -267,6 +268,8 @@ def get_page(
             except URLError:
                 if not PAGES_SOURCE_LOCATION.startswith('file://'):
                     raise
+            except TimeoutError:
+                continue
```
Comment (MasterOdin) [inline tldr.py:272]: I'm not sure we should be skipping this, in that we don't skip if we get a 500 `HTTPError`. Something about the upstream is broken in a way that we cannot connect to it, and I don't think we should continue on and potentially serve the wrong page.

**Review by MasterOdin — CHANGES_REQUESTED** [review]

File: tldr.py (hunk 1 above)
Comment (CleanMachine1) [reply tldr.py:163]: Is two seconds long enough?

Comment (Jaimepas77) [reply tldr.py]: > Is two seconds long enough?

I think so, but i wouldn't care if it was set to 10 seconds or something like that. The problem that i saw was that, if no timeout was specified, there were some occasions where the program would get stuck for more than a minute until throwing a timeout error. However I believe that 2 seconds are more than enough.

File: tldr.py (hunk 2 above)
Comment (Jaimepas77) [reply tldr.py]: I've read some more info about that and I've noticed that in fact a `URLError` (and not a `TimeoutError`) is raised when the timeout expires. Consequently I've deleted that part of the code and committed it. The behavior is that whenever a request is not answered an error message is shown to the user. Sorry for the confusion.

File: tldr.py (hunk 1 above)
Comment (MasterOdin) [reply tldr.py]: I've bumped this up to 10 seconds as I feel like while 2 is _probably_ fine, better to be a touch conservative and hopefully avoid any people complaining about things "suddenly" breaking on them.

**Pull request merged**